docs/enhancements/aws-private-link.md

```diff
@@ -0,0 +1,553 @@
+# Private clusters on AWS using PrivateLink
+
+## Summary
+
+Hive creates a private AWS cluster using OpenShift Installer and then sets up
+exclusive access to the private cluster using AWS PrivateLink. Hive creates an
+Endpoint Service in customer's AWS account for the cluster's k8s API and an
+Endpoint in Hub cluster to access the cluster's k8s API privately.
+
+## Motivation
+
+### Goals
+
+- Support creating private cluster with no access from Internet using Hive.
+
+### Non-Goals
+
+-
+
+## Proposal
+
+Create a new controller that can identify a ClusterDeployment that requires
+AWS PrivateLink access to the k8s API. This controller uses the ClusterMetadata
+and credentials for the cluster to identify NLB for the cluster, setup an
+Endpoint Service to expose the NLB to *specific* AWS role in Hub account, setup
+and Endpoint to the customer's Service in Hub account and finally make the endpoint
+DNS names available for use.
+
+Hive users the OpenShift Installer to *ONLY* create infrastructure resources and
+then waits for the controller to report the private DNS names for the Endpoint
+service. After the private DNS names are available, continue the install to wait
+for API to come up, destroy the bootstrap resources and wait for install to complete
+using the pre-existing openshift-installer subcommands. 
+
+Hive also updates the manifests for API server to accept the connections from 
+the Endpoint service DNS names at installation.
+
+### User Stories
+
+#### Story 1
+
+#### Story 2
+
+## Design Details
+
+### Private DNS names for endpoint services
+
+doc: https://docs.aws.amazon.com/vpc/latest/userguide/verify-domains.html
+
+When you create a VPC endpoint service, AWS generates endpoint-specific DNS
+hostnames that you can use to communicate with the service. These names include
+the VPC endpoint ID, the Availability Zone name and Region Name, for example, 
+vpce-1234-abcdev-us-east-1.vpce-svc-123345.us-east-1.vpce.amazonaws.com.
+
+For successful communication with API using **this** DNS name, the apiserver must
+be configured to serve the appropriate ServerCertifcate and client-go clients
+must be configured with the corresponding TrustCertificate in form of the kubeconfig.
+This configuration must be modified day-1 (during installation) because we
+need a valid working SNI configuration to exist before we can communicate with the
+apiserver to update it.
+
+So to solve the chicken-egg problem, we must use user-defined private DNS names
+for Endpoint services that are easily computed before we begin the installation.
+This requires a **public** HostedZone to allow AWS to perform Domain ownership
```

Comment:
Must assume anything public is not acceptable for the use cases on OSD.  What other options are there that do not require a public HostedZone?